Logging/LogHandler/FileLogHandler.cs

```diff
@@ -79,33 +84,42 @@ public enum LogType
         public LogLevel MinimumLevel { get; set; } = LogLevel.Debug;
 
         /// <summary>
-        /// The maximum file size in bytes before the file is rotated. Default is 10MB.
+        /// Creates a `FileLogHandler` that can write to the given file path.
         /// </summary>
-        public long RotateFileSizeLimit { get; set; } = 10L * 1024 * 1024;
-
-        /// <summary>
+        /// <param name="path">The text file to output messages to.</param>
+        /// <param name="writeMode">The <see cref="WriteMode"/> to use.</param>
+        /// <param name="rotateFileSizeLimit">
+        /// The maximum file size in bytes before the file is rotated. Default is 10MB.
+        /// If set, the file will be rotated when the next log would exceed the limit. When a file is rotated, an index is appended to it,
+        /// ex. "log.txt" is renamed "log.1.txt", and a new "log.txt" is opened. If "log.1.txt" already exists, it is first renamed
+        /// "log.2.txt", and so on, rotating through log files.
+        /// </param>
+        /// <param name="rotateFileCountLimit">
         /// The maximum number of rotated files to keep before deleting the oldest. Default is 5.
         /// Ex. If the limit is 5, up to 5 rotated files in addition to the original log file, will be kept. After reaching this limit,
         /// the next time the log file is rotated, the oldest file will be deleted. If the limit is 0, no rotated files are kept,
-        /// i.e. when the log file is rotated, it is simply erased and starts over.
+        /// i.e. when the log file is rotated, it is simply erased and started over.
         /// </summary>
-        public int RotateFileCountLimit { get; set; } = 5;
-
-        /// <summary>
-        /// Creates a `FileLogHandler` that can write to the given file path.
-        /// </summary>
-        /// <param name="path">The text file to output messages to.</param>
-        /// <param name="logType">The <see cref="LogType"/> to use.</param>
-        /// <param name="shouldRotate">
-        /// If true, the file will be rotated when the size limit is reached. When a file is rotated, it has an index appended to it,
-        /// i.e. "log.txt" is renamed "log.1.txt", and a new "log.txt" is opened. If "log.1.txt" already exists, it is first renamed
-        /// "log.2.txt", and so on, rotating through log files. Once the file count limit is reached, the oldest file is deleted.
-        /// </param>
-        public FileLogHandler(string path, LogType logType = LogType.Append, bool shouldRotate = true)
+        public FileLogHandler(
+            string path,
+            WriteMode writeMode = WriteMode.Append,
+            long? rotateFileSizeLimit = 10L * 1024 * 1024,
```

Comment:
Add 10mb comment or move to a constant and put the comment there.

Reply:
The `<param>` description above says the default is 10MB, or did you want another comment inline here?